### PR TITLE
[BugFix] avoid recursively throwing std::bad_alloc from memory limit

### PR DESCRIPTION
**Pull request by fzhedu**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by kevincai — DISMISSED** [review]

File: be/src/util/stack_util.cpp [hunk 1]
```diff
@@ -119,11 +119,20 @@ void __wrap___cxa_throw(void* thrown_exception, void* info, void (*dest)(void*))
                                      ToStringFromUnixMicros(GetCurrentTimeMicros()).c_str(), print_id(query_id).c_str(),
                                      print_id(fragment_instance_id).c_str(), exception_name.c_str(),
                                      get_stack_trace().c_str());
+            bool print_to_stderr = false;
+            if (exception_name == "std::bad_alloc") {
+                // to avoid recursively throw std::bad_alloc exception.
+                print_to_stderr = true;
```
[the remark this answers is not in the document]
Comment (trueeyu) [reply be/src/util/stack_util.cpp:125]: Now that there is already LargeAlloc logging, I still don’t quite understand what problem logging is here to solve

**Review by trueeyu — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by kangkaisen — APPROVED** [review]

**Comment by fzhedu** [comment]
run starrocks_admit_test

**Comment by fzhedu** [comment]
run starrocks_fe_unittest

**Review by kevincai — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**